bigquery/google/cloud/bigquery/table.py

```diff
@@ -473,8 +473,9 @@ def _build_resource(self):
             resource['schema'] = {
                 'fields': _build_schema_resource(self._schema)
             }
-        else:
-            raise ValueError("Set either 'view_query' or 'schema'.")
+        elif self.partitioning_type is None:
+            raise ValueError(
+                "Set either 'view_query' or 'schema' or 'partitioning_type'.")
```

Comment:
This error message is getting a little confusing. Maybe the client shouldn't be doing this kind of validation?

I think schema isn't required if autodetect is set to true, for example.